### ISSUE TITLE
emval.js break code in v3.1.15 from v3.1.14, commit #17293

### ISSUE THREAD

**Issue by michelerenzullo** [emscripten-core/emscripten#17546]
In emsdk v3.1.15 with the commit #17293 my code is broken, v3.1.14 is fine:

I have this issue in multiple projects since this commit, using embind I pass to my function as arguments a couple of pointers and I return an `emscripten::val::global("Uint8Array").new_(emscripten::typed_memory_view(suze, array_data));`

Reproduce the error:
- call a function for the first time with X Memory, **everything OK**
- we call again for the 2nd time with Y > X memory , (-sALLOW_MEMORY_GROWTH)
- it doesn't work and it returns in consol log `parameter 0 has unknown type emsc`

Source of the problem is in emval.js lines 210-211:

https://github.com/emscripten-core/emscripten/blob/70905f463cf5e16a3bd3dba92d8689e6ef971254/src/embind/emval.js#L209-L212

In the `function craftEmvalAllocator(argCount`) of my .js output I had to remove "HEAPU32" like was before in v3.1.14 or set "HEAPU8" to make it works like v3.1.14 , (this workaround doesn't work with --closure)

from 
```
      return (new Function("requireRegisteredType", "Module", "valueToHandle", "HEAPU32" , functionBody))(
          requireRegisteredType, Module, Emval.toHandle, HEAPU32);
```

to

```
      return (new Function("requireRegisteredType", "Module", "valueToHandle", "HEAPU8" /* or just leave empty like 3.1.14 */ , functionBody))(
          requireRegisteredType, Module, Emval.toHandle, HEAPU8 /* or just leave empty like 3.1.14 */);
```

**Comment by michelerenzullo** [comment]
Thanks @sbc100 Fix is in [#17556](https://github.com/emscripten-core/emscripten/pull/17556)
Hope they review and merge soon

**Comment by sbc100** [comment]
Closing as fixed since #17556 has now landed.
